plugin/src/main/java/com/playmonumenta/scriptedquests/managers/ZoneManager.java

```diff
@@ -0,0 +1,337 @@
+package com.playmonumenta.scriptedquests.managers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.util.Vector;
+
+import com.playmonumenta.scriptedquests.Plugin;
+import com.playmonumenta.scriptedquests.utils.QuestUtils;
+import com.playmonumenta.scriptedquests.zones.ZoneChangeEvent;
+import com.playmonumenta.scriptedquests.zones.ZonePropertyChangeEvent;
+import com.playmonumenta.scriptedquests.zones.ZoneLayer;
+import com.playmonumenta.scriptedquests.zones.zone.BaseZone;
+import com.playmonumenta.scriptedquests.zones.zone.Zone;
+import com.playmonumenta.scriptedquests.zones.zone.ZoneFragment;
+import com.playmonumenta.scriptedquests.zones.zonetree.BaseZoneTree;
+
+public class ZoneManager {
+	private Plugin mPlugin;
+	static BukkitRunnable playerTracker = null;
+
+	private HashMap<String, ZoneLayer> mLayers = new HashMap<String, ZoneLayer>();
+	private HashMap<String, ZoneLayer> mPluginLayers = new HashMap<String, ZoneLayer>();
+	private BaseZoneTree mZoneTree = null;
+	private Map<Player, ZoneFragment> lastPlayerZoneFragment = new HashMap<Player, ZoneFragment>();
+
+	public ZoneManager(Plugin plugin) {
+		mPlugin = plugin;
+	}
+
+	/************************************************************************************
+	 * Start of methods for use with external plugins:
+	 ************************************************************************************/
+
+	/*
+	 * Register a ZoneLayer from an external plugin.
+	 *
+	 * Returns true on success, false on failure.
+	 */
+	public boolean registerPluginZoneLayer(ZoneLayer layer) {
+		if (layer == null) {
+			return false;
+		}
+
+		String layerName = layer.getName();
+
+		if (mPluginLayers.containsKey(layerName)) {
+			return false;
+		}
+
+		mLayers.put(layerName, layer);
+		reload(mPlugin, null);
+		return true;
+	}
+
+	/*
+	 * Unregister a ZoneLayer from an external plugin.
+	 *
+	 * Returns true on success, false on failure.
+	 */
+	public boolean unregisterPluginZoneLayer(String layerName) {
+		if (!mPluginLayers.containsKey(layerName)) {
+			return false;
+		}
+
+		mPluginLayers.remove(layerName);
+		reload(mPlugin, null);
+		return true;
+	}
```

Comment:
It would allow an external plugin to register zones that track players. I figure this is how plots is likely to work in the future, unless we want to maintain the zones manually. I'm not sure how we're going to keep the zones up to date no matter which way we do it yet.

Reply:
Makes sense. Still not sure it's needed, but it can stay.